src/types.rs

```diff
@@ -247,6 +247,32 @@ pub struct TransactionDetails {
     pub confirmation_time: Option<BlockTime>,
 }
 
+impl PartialOrd for TransactionDetails {
+    fn partial_cmp(&self, other: &Self) -> Option<std::cmp::Ordering> {
+        Some(self.cmp(other))
+    }
+}
+
+impl Ord for TransactionDetails {
+    fn cmp(&self, other: &Self) -> std::cmp::Ordering {
+        // Sort by timestamp, but if there's no timestamp put it first,
+        // if timestamps are equal, compare by txid
+        match (&self.confirmation_time, &other.confirmation_time) {
+            (Some(a), Some(b)) => {
+                let cmp = a.timestamp.cmp(&b.timestamp);
```

Comment:
Maybe it would make sense to implement `Ord` also for `BlockTime`, which would first compare the height, then the timestamp, and then just call compare the entire `confirmation_time` here?

Reply:
Good idea, done